### PR TITLE
[aws] Remove unused aws.*.metrics.*.* and aws.s3.bucket.name

### PR DESCRIPTION
**Pull request by kaiyan-sheng**
## Proposed commit message

This PR is to remove unused `aws.*.metrics.*.*` and `aws.s3.bucket.name` from package-fields.yml file.
For some data streams when defining what metrics to collect from CloudWatch, specific metric names are listed. With the list of specific metrics, `aws.*.metrics.*.*` is redundant. Some data stream when defining what metrics to collect, it only specifies the namespace for CloudWatch. That way we will keep the `aws.*.metrics.*.*` just in case AWS adds more metrics.

For `aws.s3.bucket.name` it is only used in `s3_daily_storage` and `s3_request` data streams in AWS integration. Will remove the rest.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

### CODE REVIEW

File: packages/aws/data_stream/s3_daily_storage/fields/package-fields.yml [hunk 1]
```diff
@@ -7,6 +7,5 @@
         Tag key value pairs from aws resources.
     - name: s3.bucket.name
       type: keyword
-      dimension: true
```
[the remark this answers is not in the document]
Comment (kaiyan-sheng) [reply packages/aws/data_stream/s3_daily_storage/fields/package-fields.yml:10]: Not sure why this is removed hmm Definitely not intended. Will add it back!

**Review by zmoog — APPROVED** [review]
LGTM

**Review by bhapas — APPROVED** [review]
LGTM

**Pull request merged**